### PR TITLE
[RLlib] Trajectory View API (preparatory cleanup and enhancements).

### PR DESCRIPTION
**Pull request by sven1977**
This PR adds some code cleanup and enhancements (e.g. time-major option for some helper functions) in preparation for an in-the-pipeline "Trajectory View LSTM" PR.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: ci/travis/format.sh [hunk 1]
```diff
@@ -104,7 +104,7 @@ format_changed() {
              yapf --in-place "${YAPF_EXCLUDES[@]}" "${YAPF_FLAGS[@]}"
         if which flake8 >/dev/null; then
             git diff --name-only --diff-filter=ACRM "$MERGEBASE" -- '*.py' | xargs -P 5 \
-                 flake8 --inline-quotes '"' --no-avoid-escape --exclude=python/ray/core/generated/,streaming/python/generated,doc/source/conf.py,python/ray/cloudpickle/,python/ray/thirdparty_files/ --ignore=C408,E121,E123,E126,E226,E24,E704,W503,W504,W605
+                 flake8 --inline-quotes '"' --no-avoid-escape --exclude=python/ray/core/generated/,streaming/python/generated,doc/source/conf.py,python/ray/cloudpickle/,python/ray/thirdparty_files/ --ignore=C408,E121,E123,E126,E226,E24,E704,W503,W504,W605,F821
```
Comment (sven1977) [inline ci/travis/format.sh:107]: To ignore errors related to Type-annotations (those that use quotes around the actual class (b/c we want to avoid circular import errors)).

File: rllib/evaluation/postprocessing.py [hunk 2]
```diff
@@ -39,10 +39,7 @@ def compute_advantages(rollout: SampleBatch,
             processed rewards.
     """
 
-    traj = {}
-    trajsize = len(rollout[SampleBatch.ACTIONS])
-    for key in rollout:
-        traj[key] = np.stack(rollout[key])
+    rollout_size = len(rollout[SampleBatch.ACTIONS])
```
Comment (sven1977) [inline rllib/evaluation/postprocessing.py:42]: We shouldn't copy the entire batch here. Significantly speeds up postprocessing.

File: rllib/evaluation/rollout_worker.py [hunk 3]
```diff
@@ -334,8 +330,10 @@ def wrap(env):
             # Deepmind wrappers already handle all preprocessing
             self.preprocessing_enabled = False
 
-            if clip_rewards is None:
-                clip_rewards = True
+            # If clip_rewards not explicitly set to False, switch it
```
Comment (sven1977) [inline rllib/evaluation/rollout_worker.py:333]: clip_rewards has been moved into base-Policy's `postprocess_trajectory` method.

File: rllib/examples/env/multi_agent.py [hunk 4]
```diff
@@ -164,3 +165,5 @@ def step(self, action_dict):
 MultiAgentCartPole = make_multiagent("CartPole-v0")
 MultiAgentMountainCar = make_multiagent("MountainCarContinuous-v0")
 MultiAgentPendulum = make_multiagent("Pendulum-v0")
+MultiAgentStatelessCartPole = make_multiagent(
```
Comment (sven1977) [inline rllib/examples/env/multi_agent.py:168]: Added for LSTM multi-agent testing purposes.

File: rllib/utils/torch_ops.py [hunk 5]
```diff
@@ -10,11 +10,8 @@
 def explained_variance(y, pred):
     y_var = torch.var(y, dim=[0])
     diff_var = torch.var(y - pred, dim=[0])
-    min_ = torch.Tensor([-1.0])
-    return torch.max(
-        min_.to(device=torch.device("cuda"))
-        if torch.cuda.is_available() else min_,
-        1 - (diff_var / y_var))
+    min_ = torch.tensor([-1.0]).to(pred.device)
```
Comment (sven1977) [inline rllib/utils/torch_ops.py:13]: This was a bug (should only use GPU if num_gpus > 0, not whenever one is available).

File: rllib/evaluation/rollout_worker.py (hunk 3 above)
Comment (ericl) [reply rllib/evaluation/rollout_worker.py:333]: Won't this break reward clipping for policies that don't call that method?

**Review by ericl — CHANGES_REQUESTED** [review]
Why do we need the reward clipping changes?

File: rllib/evaluation/rollout_worker.py (hunk 3 above)
Comment (sven1977) [reply rllib/evaluation/rollout_worker.py:337]: True, if users don't use the build_policy methods, then yes. The thought was: reward clipping is actually a post-processing step.
Undid these changes again.

**Comment by sven1977** [comment]
@ericl

File: rllib/evaluation/rollout_worker.py (hunk 3 above)
Comment (ericl) [reply rllib/evaluation/rollout_worker.py:337]: Sounds good. Let's be careful about breaking changes like that.

**Review by ericl — APPROVED** [review]
LGTM

File: rllib/evaluation/rollout_worker.py (hunk 3 above)
Comment (sven1977) [reply rllib/evaluation/rollout_worker.py:337]: +1
